### PR TITLE
MULE-12284: Remove MultiPartPayload

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

File: integration/src/test/java/org/mule/test/integration/EventMetaDataPropagationTestCase.java [hunk 1]
```diff
@@ -57,33 +48,28 @@ public InternalEvent process(InternalEvent event) throws MuleException {
...
     private Message doProcess(InternalEvent event) throws MuleException {
-      if ("component1".equals(getLocation().getRootContainerName())) {
-        Map<String, Serializable> props = new HashMap<>();
-        props.put("stringParam", "param1");
-        props.put("objectParam", new Apple());
-        props.put("doubleParam", 12345.6);
-        props.put("integerParam", 12345);
-        props.put("longParam", (long) 123456789);
-        props.put("booleanParam", Boolean.TRUE);
+            if ("component1".equals(getLocation().getRootContainerName())) {
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/integration/EventMetaDataPropagationTestCase.java]: ?

File: integration/src/test/java/org/mule/test/integration/EventMetaDataPropagationTestCase.java [hunk 2]
```diff
@@ -57,33 +48,28 @@ public InternalEvent process(InternalEvent event) throws MuleException {
...
-      } else {
-        InternalMessage msg = (InternalMessage) event.getMessage();
-        assertEquals("param1", msg.getInboundProperty("stringParam"));
-        final Object o = msg.getInboundProperty("objectParam");
-        assertTrue(o instanceof Apple);
-        assertEquals(12345.6, 12345.6, msg.<Double>getInboundProperty("doubleParam", 0d));
-        assertEquals(12345, msg.<Integer>getInboundProperty("integerParam", 0).intValue());
-        assertEquals(123456789, msg.<Long>getInboundProperty("longParam", 0L).longValue());
-        assertEquals(Boolean.TRUE, msg.getInboundProperty("booleanParam", Boolean.FALSE));
-        assertThat(msg.getPayload().getValue(), instanceOf(DefaultMultiPartPayload.class));
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/integration/EventMetaDataPropagationTestCase.java]: can these assertions be made in the new model?

File: integration/src/test/java/org/mule/test/integration/EventMetaDataPropagationTestCase.java (hunk 1 above)
Comment (afelisatti) [inline integration/src/test/java/org/mule/test/integration/EventMetaDataPropagationTestCase.java]: ¯\_(ツ)_/¯ this is a very old test, I just removed references to multipart.

File: integration/src/test/java/org/mule/test/integration/EventMetaDataPropagationTestCase.java (hunk 2 above)
Comment (afelisatti) [inline integration/src/test/java/org/mule/test/integration/EventMetaDataPropagationTestCase.java]: I'll check

**Comment by afelisatti** [comment]
@elrodro83 turns out the `component2` flow and else clause wasn't used.

**Pull request merged**